src/sortStudents.ts

```diff
@@ -1,16 +1,65 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function findAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, mark) => sum + mark, 0) / grades.length;
+}
+
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const copiedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copiedStudents.sort((currentStudent, nextStudent) => {
+        return order === 'asc'
+          ? currentStudent[sortBy].localeCompare(nextStudent[sortBy])
+          : nextStudent[sortBy].localeCompare(currentStudent[sortBy]);
+      });
+
+      break;
```

Comment:
Let's try follow DRY , `copiedStudents.sort` repeat three times , can we make return immediately like:
```suggestion
return copiedStudents.sort((a, b) => {
  switch (sortBy) {
    case SortType.Name:
    case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
```

1. also you don't need brake after `return`
2. and with other cases the same, it make your code more readable and you use sort just once :)